modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb

```diff
@@ -51,13 +51,20 @@ def collection_resource
       end
 
       def last_refill_date_filter(resource)
-        resource = resource.sort_by do |x|
-          x.rx_rf_records?.rf_record.find do |rf_record|
-            rf_record[:dispensed_date].present?
-          end.present? || x[:dispensed_date].present?
-        end
+        resource = resource.data.sort_by do |x|
+          has_refills = x.try(:rx_rf_records).present?
+
+          last_refill_date = 30.years.ago
+          last_refill_date = x.rx_rf_records.flatten[1][:dispensed_date]  if has_refills
```

Comment:
I think this might be more clear as a ternary statement or even a full if/else

Reply:
@acrollet I had it as a full statement, but robocop failed and recommended this.

Reply:
ah ok, fair enough